### PR TITLE
Fix HDF5 PV generator regression failure

### PR DESCRIPTION
**Pull request by pford**
**Description**

* What is implemented or fixed? Mention the linked issue(s), if any.
#1259 
* How does this PR solve the issue? Give a brief summary.
The HDF5 PV image had duplicate profiles from the Hdf5Loader cache. The cache was deleted only for region id = TEMP_REGION_ID.  However, for parallel profiles each region now has a unique id decremented from the temp id, so the profile was being cached and reused when the same id was assigned later.  Now the cache is deleted if the region id <= TEMP_REGION_ID. Also removed distortion check output for matched regions, causing too much debug output; now the log only reports that the polygon approximation is being used.
* Are there any companion PRs (frontend, protobuf)?
No.
* Is there anything else that testers should know (e.g. exactly how to reproduce the issue)?
See description in issue: create PV images from identical FITS and HDF5 source images, then compare them using a LEL expression.

**Checklist**

- [x] changelog updated / ~no changelog update needed~
- [ ] e2e test passing / added corresponding fix
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] added reviewers and assignee
- [x] added ZenHub estimate, milestone, and release

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
this appears fine with some manual tests. no regression from e2e tests. @acdo2002 could you repeat your test to see if the problem is fixed?

**Review by acdo2002 — APPROVED** [review]
Yes, the regression problem solved in this PR.
The ICD-RxJS are all passed now. Once this PR merged into the dev, I will unblock the part of ICD tests.
And manual check the same PV cut in fits and hdf5 is identical, because two PV cut images subtracted are all 0.
Screen Shot 2023-05-17 at 5 02 04 PM

**Review by confluence — APPROVED** [review]

**Pull request merged**
